haskell/src/Client/Contract.hs

```diff
@@ -0,0 +1,73 @@
+-- SPDX-FileCopyrightText: 2020 tqtezos
+-- SPDX-License-Identifier: MIT
+
+module Client.Contract
+  ( stablecoinContractContent
+  , parseStablecoinContract
+  , mkInitialStorage
+  ) where
+
+import Data.FileEmbed (embedStringFile)
+import qualified Data.Map.Strict as Map
+import Michelson.Runtime (parseExpandContract)
+import Michelson.Text (MText)
+import Michelson.Typed (BigMap(BigMap))
+import qualified Michelson.Untyped as U
+import Tezos.Address (Address)
+import Util.Named ((.!))
+
+import Lorentz.Contracts.Stablecoin (Storage, mkTokenMetadata)
+
+stablecoinContractContent :: Text
+stablecoinContractContent = $(embedStringFile "../ligo/stablecoin.tz")
+
+parseStablecoinContract :: MonadThrow m => m U.Contract
+parseStablecoinContract =
+  either throwM pure $
+    parseExpandContract
+      (Just "../ligo/stablecoin.tz")
+      stablecoinContractContent
+
+mkInitialStorage :: Address -> Address -> Address -> Address -> MText -> MText -> Natural -> Storage
```

Comment:
I don't like that there are so many arguments and some of them have same types. It looks similar to `DeploymentData`, but there we have `AddressOrAlias` and here we have `Address`. What about parameterizing `DeploymentData` with `a` and passing `DeploymentData Address` here (and `DeploymentData AddressOrAlias` in another place)?

Reply:
I like that! Done.